dotCMS/src/main/java/com/dotcms/storage/ContentletMetadataAPIImpl.java

```diff
@@ -0,0 +1,384 @@
+        final String metadataBucketName = Config.getStringProperty(METADATA_GROUP_NAME,
+                DOT_METADATA);
+        for (final String fullBinaryFieldName : fullBinaryFieldNameSet) {
+            Logger.info(ContentletMetadataAPIImpl.class,"");
+            final File file           = contentlet.getBinary(fullBinaryFieldName);
+            final String metadataPath = this.getFileName(contentlet, fullBinaryFieldName);
+            if (null != file && file.exists() && file.canRead()) {
+                final Set<String> metadataFields = this.getMetadataFields(fieldMap.get(fullBinaryFieldName).id());
+                final Map<String, Object> metadataMap = this.fileStorageAPI.generateMetaData(file,
+                        new GenerateMetadataConfig.Builder()
+                            .full(true)
+                            .override(alwaysRegenerateMetadata)
+                            .cache(false)  // do not want cache on full meta
+                            .store(true)
+                            .metaDataKeyFilter(metadataKey -> metadataFields.isEmpty()
+                                    || metadataFields.contains(metadataKey))
+                            .storageKey(new StorageKey.Builder().group(metadataBucketName).path(metadataPath).storage(storageType).build())
+                            .build()
+                        );
+
+                fullMetadataMap.put(fullBinaryFieldName, metadataMap);
+            } else {
+
+                throw new IOException("The file: " + file + ", is null, does not exists or can not access");
+            }
+        }
+    }
+
+    /**
+     * based on the identifier this will give you a set of fields for the metadata generation
+     * fields are specific to the CT or preconfigured
+     * @param fieldIdentifier
+     * @return
+     */
+    @VisibleForTesting
+    Set<String> getMetadataFields (final String fieldIdentifier) {
+
+        final Optional<FieldVariable> customIndexMetaDataFieldsOpt =
+                Try.of(()->FactoryLocator.getFieldFactory().byFieldVariableKey(fieldIdentifier, BinaryField.INDEX_METADATA_FIELDS)).getOrElse(Optional.empty());
+
+        final Set<String> metadataFields = customIndexMetaDataFieldsOpt.isPresent()?
+                new HashSet<>(Arrays.asList(customIndexMetaDataFieldsOpt.get().value().split(StringPool.COMMA))):
+                this.getConfiguredMetadataFields();
+
+        Logger.info(ContentletMetadataAPIImpl.class,
+                () -> String.format(" `%s` has these fields: `%s` ", fieldIdentifier, String
+                        .join(",", metadataFields)));
+        return metadataFields;
+    }
+
+    /**
+     * Reads INDEX_METADATA_FIELDS for  pre-configured metadata fields
+     * @return
+     */
+    private Set<String> getConfiguredMetadataFields(){
+
+        final java.lang.String configFields = Config.getStringProperty("INDEX_METADATA_FIELDS", null);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'java.lang.String' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)